### PR TITLE
Add recipe auto-scroll-mode

### PR DESCRIPTION
**Pull request by stardiviner**
### Brief summary of what the package does

Auto scroll buffer lines without hand touching.

### Direct link to the package repository

https://github.com/stardiviner/auto-scroll-mode

### Your association with the package

maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I have confirmed some of these without doing them

### CODE REVIEW

**Comment by conao3** [comment]
Hi!
I'm not a MELPA maintainer, but here's some feedback for this package.
I hope below those comments help you and your package.

0. Please open PR per package. Maybe you need rebase `origin/master`?

**Comment by stardiviner** [comment]
@conao3 You're right, I forgot create a clean branch. I will force push this branch later.

**Comment by conao3** [comment]
0. Please resolve below package-lint/check-doc/byte-compiler warnings

```diff
new file   Cask
@@ -0,0 +1,4 @@
+(source gnu)
+(source melpa)
+
+(package-file "auto-scroll-mode.el")
```

```
$ cask; cask build
Loading package information... done
Package operations: 1 installs, 0 removals
  - Installing [ 1/1] emacs (25)... already present
Compiling /home/conao/dev/forks/auto-scroll-mode/auto-scroll-mode.el...

In auto-scroll-start:
auto-scroll-mode.el:75:9:Warning: ‘goto-line’ is for interactive use only; use
    ‘forward-line’ instead.

In end of data:
auto-scroll-mode.el:139:1:Warning: the following functions are not known to be
    defined: incf, decf
```

**Comment by stardiviner** [comment]
Ok, updated, should be fixed now. Thanks @conao3

File: recipes/auto-scroll-mode [hunk 1]
```diff
@@ -0,0 +1 @@
+(auto-scroll-mode :repo "stardiviner/auto-scroll-mode" :fetcher github :files ("auto-scroll-mode.el"))
```
Comment (riscy) [inline recipes/auto-scroll-mode]: Looks like you can get away with the default recipe?
```suggestion
(auto-scroll-mode :repo "stardiviner/auto-scroll-mode" :fetcher github)
```

**Comment by stardiviner** [comment]
Ok, fixed now, I barely know how to category extension functionality in `finder-known-keywords`. Keep pedantic on this is good for users. You're right.

All minor typo and issues solved.

**Pull request merged**